vernac/g_vernac.mlg

```diff
@@ -296,10 +302,40 @@ GRAMMAR EXTEND Gram
            univdecl_extensible_constraints = snd cs } }
     ] ]
   ;
+  variance:
+    [ [ "+" -> { Univ.Variance.Covariant }
+      | "=" -> { Univ.Variance.Invariant }
+      | "*" -> { Univ.Variance.Irrelevant }
+    ] ]
+  ;
+  variance_identref:
+    [ [ id = identref -> { (id, None) }
+      | test_variance_ident; v = variance; id = identref -> { (id, Some v) }
+        (* We need this test to help the parser avoid the conflict
+           between "+" before ident (covariance) and trailing "+" (extra univs allowed) *)
```

Comment:
Maybe we should reconsider "+" for extra declarations and use ".." instead.

Reply:
>Maybe we should reconsider "+" for extra declarations and use ".." instead.

No, the lookahead works fine so we don't need to bread backward compat.